pkg/cache/clusterqueue.go

```diff
@@ -706,16 +715,17 @@ func (c *ClusterQueue) dominantResourceShare(wlReq FlavorResourceQuantities, m i
 
 	var drs int64 = -1
 	var dRes corev1.ResourceName
+	lendable := c.Cohort.Lendable()
 	for rName, b := range borrowing {
-		if lendable := c.Cohort.Lendable[rName]; lendable > 0 {
-			ratio := b * 1000 / lendable
+		if lr := lendable[rName]; lr > 0 {
```

Comment:
why did the implementation change?
If there was a bug, please open a separate PR for it with accompanying unit tests.

Reply:
The `c.Cohort.Lendable` calculated by `Snapshot()`. To do not execute this method on each time when we need to execute `dominantResourceShare` function this logic moved to another helper that calculating cohort lendable.